### PR TITLE
Use shared NS HAL init code

### PR DESCRIPTION
**Pull request by kjbracey-arm**
For RTOS builds, use the shared NS HAL initialisation code in
nanostack-hal-cmsis-rtos. No change for non-RTOS builds.

Moving the init code out of this component allows the NS HAL to
be used by other components whether or not Nanostack itself
is in use.

### CODE REVIEW

**Comment by kjbracey-arm** [comment]
@artokin, @yogpan01, @SeppoTakalo, @c1728p9, @geky

File: source/mesh_system.c [hunk 1]
```diff
@@ -64,17 +64,20 @@ static void mesh_system_heap_error_handler(heap_fail_t event)
 void mesh_system_init(void)
 {
     if (mesh_initialized == false) {
+#ifndef YOTTA_CFG
+        ns_hal_init(app_stack_heap, MESH_HEAP_SIZE,
+                    mesh_system_heap_error_handler, NULL);
+        eventOS_scheduler_mutex_wait();
+        net_init_core();
+        eventOS_scheduler_mutex_release();
```
Comment (artokin) [inline source/mesh_system.c]: Where socket adaptation layer (ns_sal_init_stack) will be initialized in non yotta builds?

Comment (kjbracey-arm) [inline source/mesh_system.c]: The existing "central registration" SAL is not present, so there's no equivalent initialisation. This stack startup is called by a "standalone" socket/interface adaptation.

Comment (artokin) [inline source/mesh_system.c]: ok,  then it looks good.

**Comment by yogpan01** [comment]
test this please

**Pull request merged**